client-misk-jooq/src/main/kotlin/app/cash/backfila/client/misk/jooq/internal/JooqBackfillOperator.kt

```diff
@@ -0,0 +1,145 @@
+import app.cash.backfila.client.misk.jooq.JooqBackfill
+import app.cash.backfila.client.misk.spi.BackfilaParametersOperator
+import app.cash.backfila.client.misk.spi.BackfillOperator
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeRequest
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeResponse
+import app.cash.backfila.protos.clientservice.KeyRange
+import app.cash.backfila.protos.clientservice.PrepareBackfillRequest
+import app.cash.backfila.protos.clientservice.PrepareBackfillResponse
+import app.cash.backfila.protos.clientservice.PrepareBackfillResponse.Partition
+import app.cash.backfila.protos.clientservice.RunBatchRequest
+import app.cash.backfila.protos.clientservice.RunBatchResponse
+import com.google.common.collect.Streams
+import com.squareup.backfila.client.base.jooq.CompoundKeyComparer
+import okio.ByteString
+import java.util.stream.Collectors
+
+/**
+ * @param <K> the type of key the backfill iterates over.
+ * @param <Param> backfill parameters
+ */
+class JooqBackfillOperator<K, Param : Any> internal constructor(
+  override val backfill: JooqBackfill<K, Param>,
+  private val parametersOperator: BackfilaParametersOperator<Param>
+) : BackfillOperator {
+
+  override fun name(): String = backfill.javaClass.name
+
+  override fun prepareBackfill(request: PrepareBackfillRequest): PrepareBackfillResponse {
+    Validator.validateRange(request.range, backfill.keySerializer)
+
+    backfill.prepareAndValidateBackfill(
+      parametersOperator.constructBackfillConfig(
+        request.parameters, request.dry_run
+      )
+    )
+
+    val partitions = backfill.shardedTransacterMapBackfill.keys.map { partitionName ->
+      Partition.Builder()
+        .partition_name(partitionName)
+        .backfill_range(computeOverallRange(partitionName, request))
+        .estimated_record_count(null)
+        .build()
+    }
+
+    return PrepareBackfillResponse.Builder().partitions(partitions).build()
+  }
+
+  override fun getNextBatchRange(request: GetNextBatchRangeRequest): GetNextBatchRangeResponse {
+    check(request.compute_count_limit > 0) { "batch limit must be > 0" }
+
+    if (request.backfill_range.start == null) {
+      // This partition never had any data, stop it immediately.
+      return GetNextBatchRangeResponse.Builder()
+        .batches(emptyList())
+        .build()
+    }
+
+    return backfill.inTransactionReturning(
+      "JooqBackfill#getNextBatchRange",
+      request.partition_name
+    ) { dslContext ->
+      GetNextBatchRangeResponse.Builder()
+        .batches(
+          Streams.stream(BatchRangeIterator(backfill, dslContext, request))
+            .limit(request.compute_count_limit)
+            .collect(Collectors.toList())
+        )
+        .build()
+    }
+  }
+
+  override fun runBatch(request: RunBatchRequest): RunBatchResponse {
+    val keysInRange = backfill.inTransactionReturning(
+      "JooqBackfill#runBatch", request.partition_name
```

Comment:
Couldn't you add the backfill short name here?

Reply:
Yep.